File: tests/fixtures/vulnerable/frontend_secret.py
"""Vulnerable: secret keys exposed in frontend/browser-delivered code."""

# Bad: Stripe secret key in a file that would be bundled to the browser
STRIPE_SECRET_KEY = "SATURNDAY_TEST_SK_TOKEN"

# Bad: generic API secret in frontend config
API_SECRET = "secret_key_abc123def456ghi789"

# Bad: private key literal in frontend JS bundle context
PRIVATE_KEY = "abcdef1234567890abcdef1234567890"


def get_stripe_client():
    import stripe
    # Bad: secret key hardcoded inline
    stripe.api_key = "SATURNDAY_TEST_SK_TOKEN"
    return stripe

File: tests/fixtures/vulnerable/frontend_secret.ts
// Vulnerable: secret keys exposed in frontend/browser-delivered code

// Bad: Stripe secret key on the client
const STRIPE_SECRET_KEY = "SATURNDAY_TEST_SK_TOKEN";

// Bad: direct Stripe charge from frontend (should be server-side)
async function chargeCustomer(amount: number) {
  const response = await fetch("https://api.stripe.com/v1/charges", {
    method: "POST",
    headers: {
      "Authorization": `Bearer ${STRIPE_SECRET_KEY}`,
    },
    body: JSON.stringify({ amount }),
  });
  return response.json();
}

// Bad: generic secret exposed
const API_SECRET = "secret_abc123def456";
const ADMIN_TOKEN = "tok_live_xxxxxxxxxxxxxxxx";

export { chargeCustomer };
